### ISSUE TITLE
About the pytorch implementation and the Replacement of the linear layer 

### ISSUE THREAD

**Issue by youthfrog** [SRKH/S4NN#2]
hello, I'm trying to implement your architecture with the linear layer you mentioned in paper(Spike time displacement based error backpropagation in convolutional spiking neural networks) backward method in pytorch. [https://github.com/youthfrog/spikng-model-learning](url)
However, maybe because of my poor implementation or some other reasons, the accuracy is not good. 
I would like to know if my idea is possible and give me some advice.
Thanks a lot.

**Comment by SRKH** [comment]
Hi
If I correctly understood your question, you are trying to implement S4NN with a piecewise linear neuron model. You should note that the Idea of dt_i/dv_i=-1 does not work here anymore, Instead, you should use the derivation in Eq.9 of the paper "Spike time displacement based error backpropagation in convolutional spiking neural networks". Also, you should use Eq.8 for the derivation of dv_i/dw_ij.

**Comment by youthfrog** [comment]
Thanks a lot. I'll try again.
